packages/cst-dts-gen/src/api.ts

```diff
@@ -0,0 +1,30 @@
+import { BaseParser } from "chevrotain"
+import { buildModel } from "./model"
+import { genDts, GenDtsOptions } from "./generate"
+
+const defaultOptions: GenDtsOptions = {
+  includeTypes: true,
+  includeVisitorInterface: true,
+  visitorInterfaceName: "ICstNodeVisitor"
+}
+
+export function generateCstDts(
+  parser: BaseParser,
+  options?: GenerateDtsOptions
+): string {
+  const effectiveOptions: GenDtsOptions = {
+    ...defaultOptions,
+    ...options
+  }
+
+  const productions = parser.getGAstProductions()
+  const model = buildModel(productions)
+
+  return genDts(model, effectiveOptions)
+}
+
+export type GenerateDtsOptions = {
```

Comment:
we have this declared in api.d.ts, why do we need it declared again?
A type in TypeScript disappears in the .js output

Reply:
good point